src/encoded/static/components/static-pages/placeholders/MdSortableTable.js

```diff
@@ -0,0 +1,639 @@
+                        typeof (column.ascSortFunction) == "function") {
+                        sortedData = column.ascSortFunction(sortedData, key);
+                    } else {
+                        sortedData = this.ascSortData(sortedData, key);
+                    }
+                    break;
+            }
+        }
+        return sortedData;
+    }
+
+    ascSortData(data, key) {
+        return this.sortDataByKey(data, key, ((a, b) => {
+            if (Utils.parseFloatable(a) && Utils.parseFloatable(b)) {
+                a = Utils.parseIfFloat(a);
+                b = Utils.parseIfFloat(b);
+            }
+            if (a >= b) {
+                return 1;
+            } else if (a < b) {
+                return -1;
+            }
+        }).bind(this));
+    }
+
+    descSortData(data, key) {
+        return this.sortDataByKey(data, key, ((a, b) => {
+            if (Utils.parseFloatable(a) && Utils.parseFloatable(b)) {
+                a = Utils.parseIfFloat(a);
+                b = Utils.parseIfFloat(b);
+            }
+            if (a <= b) {
+                return 1;
+            } else if (a > b) {
+                return -1;
+            }
+        }).bind(this));
+    }
+
+    sortDataByKey(data, key, fn) {
+        const clone = [...data];
+
+        return clone.sort((a, b) =>
+            fn(a[key], b[key])
+        );
+    }
+
+    onStateChange(index) {
+        const sortings = this.state.sortings.map(((sorting, i) => {
+            if (i == index)
+                sorting = this.nextSortingState(sorting);
+
+            return sorting;
+        }).bind(this));
+
+        this.setState({
+            sortings
+        });
+    }
+
+    nextSortingState(state) {
+        let next;
+        switch (state) {
+            case "both":
+                next = "desc";
+                break;
+            case "desc":
+                next = "asc";
+                break;
+            case "asc":
+                next = "both";
+                break;
+        }
+        return next;
+    }
+
+    setHeaderWidths(widths){
+        if (!Array.isArray(widths)) throw new Error('widths is not an array');
+        this.setState({ 'widths' : widths });
+    }
+
+    render() {
+        const { data, columns, iconAsc, iconDesc, iconBoth } = this.props;
+        const { sortings, widths } = this.state;
+        _.each(columns, function (it, idx) { it.width = widths[idx]; });
+
+        const sortedData = this.sortData(data, sortings);
+        const fullRowWidth = widths.reduce((a, b) => a + b, 0);
+
+        return (
+            <div className="markdown-table-outer-container">
+                <div className="markdown-table-container">
+                    <SortableTableHeader
+                        {...{ columns, sortings, iconDesc, iconAsc, iconBoth, setHeaderWidths: this.setHeaderWidths, headerColumnWidths: widths }}
+                        onStateChange={this.onStateChange.bind(this)} />
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JSX props should not use .bind() (react/jsx-no-bind)](https://app.codacy.com/app/4dn/fourfront/pullRequest?prid=4682591)